src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -0,0 +1,296 @@
+{
+    //
+    // The class maintains the state of the authentication process and the security context.
+    // It encapsulates security context and does the real work in authentication and
+    // user data encryption with NEGO SSPI package.
+    //
+    // This is part of the NegotiateStream PAL.
+    //
+    internal static class NegotiateStreamPal
+    {
+        // value should match the Windows sspicli NTE_FAIL value
+        // defined in winerror.h
+        private const int NTE_FAIL = unchecked((int)0x80090020);
+
+        internal static IIdentity GetIdentity(NTAuthentication context)
+        {
+            Debug.Assert(!context.IsServer, "GetIdentity: Server is not supported");
+
+            string name = context.Spn;
+            string protocol = context.ProtocolName;
+
+            return new GenericIdentity(name, protocol);
+
+        }
+
+        internal static string QueryContextAssociatedName(SafeDeleteContext securityContext)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        internal static string QueryContextAuthenticationPackage(SafeDeleteContext securityContext)
+        {
+            return NegotiationInfoClass.Kerberos;
+        }
+
+        internal static int QueryMaxTokenSize(string package)
+        {
+            // This value is not used on Unix
+            return 0;
+        }
+
+        internal static string QueryContextClientSpecifiedSpn(SafeDeleteContext securityContext)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        internal static SafeFreeCredentials AcquireDefaultCredential(string package, bool isServer)
+        {
+            return AcquireCredentialsHandle(package, isServer, new NetworkCredential(string.Empty, string.Empty, string.Empty));
+        }
+
+        internal static SafeFreeCredentials AcquireCredentialsHandle(string package, bool isServer, NetworkCredential credential)
+        {
+            if (isServer)
+            {
+                throw new PlatformNotSupportedException(SR.net_nego_server_not_supported);
+            }
+
+            SafeFreeCredentials outCredential;
+            bool isNtlm = string.Equals(package, NegotiationInfoClass.NTLM);
+
+            if (isNtlm)
+            {
+                throw new PlatformNotSupportedException(SR.net_nego_ntlm_not_supported);
+            }
+
+            if (string.IsNullOrEmpty(credential.UserName) || string.IsNullOrEmpty(credential.Password))
+            {
+                // In client case, equivalent of default credentials is to use previous,
+                // cached Kerberos TGT to get service-specific ticket.
+                outCredential = new SafeFreeNegoCredentials(string.Empty, string.Empty, string.Empty);
+            }
+            else
+            {
+                outCredential = new SafeFreeNegoCredentials(credential.UserName, credential.Password, credential.Domain);
+            }
+            return outCredential;
+
```

Comment:
Nit: unnecessary blank line.  Separately, what you have is fine, but you could if you wanted condense it via a ternary operator, e.g.
```C#
return string.IsNullOrEmpty(credential.UserName) || string.IsNullOrEmpty(credential.Password) ?
    new SafeFreeNegoCredentials(string.Empty, string.Empty, string.Empty) :
    new SafeFreeNegoCredentials(credential.UserName, credential.Password, credential.Domain);
```